File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/07. CLIMOVIL/cliente_movil_restful_java_conuni_gr06/app/src/main/java/ec/edu/monster/servicio/ServicioMasa.java
package ec.edu.monster.servicio;

import ec.edu.monster.util.ClienteRest;
import org.json.JSONObject;

public class ServicioMasa {

    public double kilogramosALibras(double kilogramos) throws Exception {
        return invocar("/masa/kilogramos-a-libras", kilogramos);
    }

    public double gramosAOnzas(double gramos) throws Exception {
        return invocar("/masa/gramos-a-onzas", gramos);
    }

    public double toneladasAKilogramos(double toneladas) throws Exception {
        return invocar("/masa/toneladas-a-kilogramos", toneladas);
    }

    public double librasAOnzas(double libras) throws Exception {
        return invocar("/masa/libras-a-onzas", libras);
    }

    public double miligramosAGramos(double miligramos) throws Exception {
        return invocar("/masa/miligramos-a-gramos", miligramos);
    }

    private double invocar(String ruta, double valor) throws Exception {
        JSONObject respuesta = ClienteRest.get(ruta, valor);
        return respuesta.getDouble("valor");
    }
}

File: SIN BDD/RESTFULL_DOTNET_CONUNI_GR06/08. SERVIDOR/servidor_rest_dotnet_conuni_gr06/src/Servicio/ServicioMasa.cs
namespace Ec.Edu.Monster.Servicio;

public class ServicioMasa
{
    public double KilogramosALibras(double kilogramos) => kilogramos * 2.20462;

    public double GramosAOnzas(double gramos) => gramos * 0.03527396;

    public double ToneladasAKilogramos(double toneladas) => toneladas * 1000;

    public double LibrasAOnzas(double libras) => libras * 16;

    public double MiligramosAGramos(double miligramos) => miligramos / 1000;
}
